yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -60,6 +60,7 @@ abstract class ExternalCall extends Instruction {
     const l2Gas = memory.get(gasOffset).toNumber();
     const daGas = memory.getAs<Field>(gasOffset + 1).toNumber();
     const functionSelector = memory.getAs<Field>(this.temporaryFunctionSelectorOffset).toFr();
+    const callType = this.type === 'STATICCALL' || context.environment.isStaticCall ? 'STATICCALL' : 'CALL';
```

Comment:
Maybe it's clearer to have:

```
const callType = context.environment.isStaticCall ? 'STATICCALL' : this.type;
```

Reply:
Please add as well

`// If we are already in a static call, we propagate the environment.` or similar